File: Python/Selection sort.py
import sys
"""
The logic is to find minimum number n times and put it in another array.
Once you find minimum of the array replace it with MaxInt after putting it in sorted array.
"""
def selectionSort(lst,n):   
    sortedlst=[0]*n
    for i in range(n):
        index=i
        mymin=lst[i]
        for j in range(n):
            if lst[j]<mymin:
                mymin=lst[j]
                index=j
        sortedlst[i]=lst[index]
        lst[index]=sys.maxsize      #Maximum size offered by machine
    return sortedlst

lst=[1,5,2,19,12,11,10]
print(selectionSort(lst,len(lst)))

File: Java/SelectionSort.java
//The logic is to select the minimum from the array and put this element in another array.
//After putting this element to another array we replace it my some max value 
//We repear above procedure for n times and we get the sorted array

class SelectionSort
{
	public static void main(String args[])
	{
		int arr[]={10,1,5,18,11,12};
		int n=arr.length;
		int sorted[]=new int[n];
		for(int i=0;i<n;i++)
		{
			int minimum=arr[0];
			int index=0;
			for(int j=0;j<n;j++)
				if (arr[j]<minimum)
				{
					minimum=arr[j];
					index=j;
				}
			sorted[i]=arr[index];
			arr[index]=Integer.MAX_VALUE;
			//Print.array(arr,n);
		}
		Print.array(sorted,n);
	}
	
}
class Print
{
	static void array(int arr[],int n)
	{
		for(int i=0;i<n;i++)
		{
			System.out.print(arr[i]);
			System.out.print(" ");
		}
		System.out.println();

	}
}